### PR TITLE
feat(Communication): Make ERPNext compatible with new Communication Structure

### PR DESCRIPTION
**Pull request by hrwX**
Please refer the following PR [#7263](https://github.com/frappe/frappe/pull/7263)
Do not merge without [#7263](https://github.com/frappe/frappe/pull/7263)

### CODE REVIEW

File: erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py [hunk 1]
```diff
@@ -35,22 +35,20 @@ def get_data(filters):
 
 	for lead in frappe.get_all('Lead', fields = ['name', 'lead_name', 'company_name'], filters=lead_filters):
 		data = frappe.db.sql("""
-			select 
-				`tabCommunication`.reference_doctype, `tabCommunication`.reference_name, 
+			select
```
Comment (Anurag810) [inline erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py]: Complex query. Inside loop ??  Have  you tried to optimize it.

**Review by Anurag810 — DISMISSED** [review]

File: erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py (hunk 1 above)
Comment (hrwX) [reply erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py]: Please refer the following PR [#7263](https://github.com/frappe/frappe/pull/7263)
The query is to make it compatible with new communication structure

File: erpnext/hr/report/daily_work_summary_replies/daily_work_summary_replies.py [hunk 2]
```diff
@@ -41,13 +41,20 @@ def get_data(filters):
...
-				['communication_type', '=', 'Communication'],
-				['sent_or_received', '=', 'Received']],
-			order_by='creation asc')
+
+	replies = frappe.db.sql("""
+			select `tabCommunication`.content, `tabCommunication`.text_content, `tabCommunication`.sender
+			from `tabCommunication`
+			inner join `tabDynamic Link`
+			on `tabCommunication`.name=`tabDynamic Link`.parent where
+			`tabDynamic Link`.link_doctype in %(daily_summary_emails)s and
```
Comment (netchampfaris) [inline erpnext/hr/report/daily_work_summary_replies/daily_work_summary_replies.py]: `link_doctype = 'Daily Work Summary' and link_name in %(daily_summary_emails)s`

File: erpnext/public/js/communication.js [hunk 3]
```diff
@@ -8,15 +8,22 @@ frappe.ui.form.on("Communication", {
...
+		if(issue_counter !== 0){
 			frm.add_custom_button(__("Issue"), () => {
 				frappe.confirm(__(confirm_msg, [__("Issue")]), () => {
 					frm.trigger('make_issue_from_communication');
 				})
 			}, "Make");
 		}
 
-		if(!in_list(["Lead", "Opportunity"], frm.doc.reference_doctype)) {
+		if(!in_list(["Lead", "Opportunity"], frm.doc.dynamic_link)) {
```
Comment (netchampfaris) [inline erpnext/public/js/communication.js]: frm.doc.dynamic_links is an array, this won't work

File: erpnext/public/js/communication.js [hunk 4]
```diff
@@ -8,15 +8,22 @@ frappe.ui.form.on("Communication", {
 
 	setup_custom_buttons: (frm) => {
 		let confirm_msg = "Are you sure you want to create {0} from this email";
-		if(frm.doc.reference_doctype !== "Issue") {
+		var issue_counter = 0;
+		for(var idx in frm.doc.dynamic_link) {
```
Comment (netchampfaris) [inline erpnext/public/js/communication.js]: `frm.doc.dynamic_links`

**Review by netchampfaris — DISMISSED** [review]

File: erpnext/hr/doctype/daily_work_summary/daily_work_summary.py [hunk 5]
```diff
@@ -46,13 +46,20 @@ def get_message_details(self):
...
 			self.daily_work_summary_group)
 
-		replies = frappe.get_all('Communication',
-			fields=['content', 'text_content', 'sender'],
-			filters=dict(reference_doctype=self.doctype,
-				reference_name=self.name,
-				communication_type='Communication',
-				sent_or_received='Received'),
-			order_by='creation asc')
+		replies = frappe.db.sql("""
```
Comment (surajshetty3416) [inline erpnext/hr/doctype/daily_work_summary/daily_work_summary.py]:
```suggestion
		replies = frappe.db.sql("""
			SELECT `tabcommunication`.content, 
				`tabcommunication`.text_content, 
				`tabcommunication`.sender 
			FROM `tabcommunication` 
				INNER JOIN `tabdynamic link` 
					ON `tabcommunication`.name = `tabdynamic link`.parent 
			WHERE `tabdynamic link`.link_doctype = '%(doctype)s' 
				AND `tabdynamic link`.link_name = '%(name)s' 
				AND `tabcommunication`.communication_type = 'Communication' 
				AND `tabcommunication`.sent_or_received = 'Received' 
			ORDER  BY `tabcommunication`.creation ASC 
		""", {
			"doctype": self.doctype,
			"name": self.name
		}, as_dict=True)
```
@hrwX can we aim such formatting as it makes the code much more readable.

Also, GitHub recognizes this kind of formatting and it gives better highlighting. This makes the review process easier.

**Review by surajshetty3416 — DISMISSED** [review]

File: erpnext/public/js/communication.js (hunk 3 above)
Comment (hrwX) [reply erpnext/public/js/communication.js]: fixed

File: erpnext/hr/report/daily_work_summary_replies/daily_work_summary_replies.py (hunk 2 above)
Comment (hrwX) [reply erpnext/hr/report/daily_work_summary_replies/daily_work_summary_replies.py]: fixed

File: erpnext/hr/doctype/daily_work_summary/daily_work_summary.py (hunk 5 above)
Comment (hrwX) [reply erpnext/hr/doctype/daily_work_summary/daily_work_summary.py]: formatted

File: erpnext/public/js/communication.js (hunk 4 above)
Comment (hrwX) [reply erpnext/public/js/communication.js]: fixed

**Pull request closed without merging**